protobuf-converter/src/main/java/io/confluent/connect/protobuf/ProtobufData.java

```diff
@@ -1288,20 +1287,25 @@ public void put(String messageName, SchemaBuilder builder) {
    * Class that holds the context for performing {@code fromConnectSchema}
    */
   private static class FromConnectContext {
-    private final Set<String> structNames;
+    private final Map<String, Set<String>> messageNames;
     private int defaultSchemaNameIndex = 0;
 
     public FromConnectContext() {
-      this.structNames = new HashSet<>();
+      this.messageNames = new HashMap<>();
     }
 
-    public boolean contains(String structName) {
-      return structName != null ? structNames.contains(structName) : false;
+    public boolean contains(String parent, String child) {
```

Comment:
Need some help to get familiar with the context. I assume the test covers this code to deal with nested refs. However, the newly added `getName()`s are for the connect use case, right? Could the names be null (cause the fields look like optional)? But even if they are null, there shouldn't be any NPE thrown.

Reply:
The message name should not be null, if that's what you're asking.